Store start_time/end_time for Backups with microseconds

AVMusorin opened the pull request:
Before start_time and end_time can't be used to analyze backup/restore
order if these backups were launched with high rate.

Storing microseconds should improve post-analysis with system.backups and
system.backup_log

### Changelog category (leave one):
- Improvement


### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Store start_time/end_time for Backups with microseconds

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

<!---
Directly edit documentation source files in the "docs" folder with the same pull-request as code changes

or

Add a user-readable short description of the changes that should be added to docs.clickhouse.com below.

At a minimum, the following information should be added (but add more as needed).
- Motivation: Why is this function, table engine, etc. useful to ClickHouse users?

- Parameters: If the feature being added takes arguments, options or is influenced by settings, please list them below with a brief explanation.

- Example use: A query or command.
-->

AVMusorin committed:
tests: replace NamedTuple with dataclass for BackupInfo

AVMusorin committed:
Store start_time/end_time for Backups with microseconds

Before start_time and end_time can't be used to analyze backup/restore
order if these backups were launched with high rate.

Storing microseconds should improve post-analysis with system.backups
and
system.backup_log

azat commented on src/Backups/BackupOperationInfo.h:
```diff
@@ -59,8 +59,8 @@ struct BackupOperationInfo
 
-    std::chrono::system_clock::time_point start_time;
-    std::chrono::system_clock::time_point end_time;
+    UInt64 start_time = 0;
+    UInt64 end_time = 0;
```
@AVMusorin can you please reflect that the precision is microseconds (and also in all the callers):

```suggestion
    UInt64 start_time_us = 0;
    UInt64 end_time_us = 0;
```

AVMusorin replied:
Yes. Done.

AVMusorin commented:
I will fix failed `test_backup_restore_new/test_cancel_backup.py::test_cancel_backup`

azat commented:
>I will fix failed test_backup_restore_new/test_cancel_backup.py::test_cancel_backup

Thank you!

Can you also make `test_backup_log/test.py::test_backup_log` non flaky in case of it runs multiple times? https://s3.amazonaws.com/clickhouse-test-reports/75929/3e99fad93418e12a5e74df4a421801e03525cb99/integration_tests__asan__flaky_check_.html

AVMusorin commented:
> Can you also make test_backup_log/test.py::test_backup_log non flaky in case of it runs multiple times?

Yes

AVMusorin committed:
Fix flaky test_backup_log

With IF NOT EXISTS/IF EXISTS statements test should be more stable for
multiple runs.

Add missed SYNC for drop table.

azat commented:
>[Integration tests (asan, flaky check)](https://s3.amazonaws.com/clickhouse-test-reports/json.html?PR=75929&sha=9892424e1632cd079e691e9d89f6daeb028bcad7&name_0=PR&name_1=Integration%20tests%20%28asan%2C%20flaky%20check%29) — Failed tests found: error: 0, passed: 1006, fail: 4, skipped: 15, brok

```
E           helpers.client.QueryRuntimeException: Client failed! Return code: 86, stderr: Received exception from server (version 25.2.1):
E           Code: 598. DB::Exception: Received from 172.16.1.2:9000. DB::Exception: Backup File('/backups/test_backup/') already exists. Stack trace:
E           
```

azat committed:
tests: make test_backup_log "idempotent"

AVMusorin commented:
@azat PTAL, probably we can finish this PR

azat commented:
>[Stress test (asan)](https://s3.amazonaws.com/clickhouse-test-reports/json.html?PR=75929&sha=37bed31a44f47851711ef4bfe1a41ae8a351de6d&name_0=PR&name_1=Stress%20test%20%28asan%29) — Killed by signal (in clickhouse-server.log)

Known issue
